flink-connector-cassandra/src/main/java/org/apache/flink/connector/cassandra/source/split/CassandraSplit.java

```diff
@@ -48,6 +52,30 @@ public CassandraSplitState toSplitState() {
         return new CassandraSplitState(new HashSet<>(ringRanges), splitId());
     }
 
+    public void serialize(ObjectOutputStream objectOutputStream) throws IOException {
+        objectOutputStream.writeInt(ringRanges.size());
+        for (RingRange ringRange : ringRanges) {
+            objectOutputStream.writeObject(ringRange.getStart());
+            objectOutputStream.writeObject(ringRange.getEnd());
+        }
+    }
+
+    public static CassandraSplit deserialize(ObjectInputStream objectInputStream)
+            throws IOException {
+        try {
+            final int nbRingRanges = objectInputStream.readInt();
+            Set<RingRange> ringRanges = new HashSet<>(nbRingRanges);
```

Comment:
note: Passing npRingRanges won't have the desired effect of avoiding allocations, since the backing map is resized when the map is filled to a certain load. In fact you've guaranteed that the map will be resized at least once.

Reply:
resolving that there are no more ringRanges set